lib/frontend.ml

```diff
@@ -163,7 +164,12 @@ module Make(C: S.CONFIGURATION with type 'a io = 'a Lwt.t) = struct
       let pages = Io_page.pages num in
       List.iter
         (fun (gref, page) ->
-           let id = gref mod (1 lsl 16) in
+           let rec next () =
+             let id = nf.rx_id in
+             nf.rx_id <- (succ nf.rx_id) mod (1 lsl 16) ;
+             if Hashtbl.mem nf.rx_map id then next () else id
```

Comment:
Note: The argument to `Hashtbl.create` is only the initial size. It will grow as more entries are added (although it could certainly start larger). Normally, the `mem` call will only happen once, as it's very unlikely that an ID will still be in use after 64k messages.